Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/Error.hs

```diff
@@ -33,6 +33,20 @@ instance IsHTTPError RatingError where
 
 instance IsAPIError RatingError
 
+data RideError
+  = InvalidRideRequest
+  deriving (Eq, Show, IsBecknAPIError)
+
+instanceExceptionWithParent 'HTTPException ''RideError
+
+instance IsBaseError RideError
+
+instance IsHTTPError RideError where
+  toErrorCode InvalidRideRequest = "INVALID_RIDE_START_REQUEST"
+  toHttpCode InvalidRideRequest = E400 -- should we keep it 500?
```

Comment:
400 and 500 depends on if error is coming because of user input or backend issue